src/MizarData/include/MizarData/SamplingWithFrameTrackComparisonReport.h

```diff
@@ -70,11 +79,51 @@ class SamplingCounts {
   uint64_t total_callstacks_{};
 };
 
-struct SamplingWithFrameTrackComparisonReport {
-  SamplingCounts baseline_sampling_counts;
-  SamplingCounts comparison_sampling_counts;
-  orbit_client_data::ScopeStats baseline_frame_track_stats;
-  orbit_client_data::ScopeStats comparison_frame_track_stats;
+class SamplingWithFrameTrackComparisonReport {
+ public:
+  explicit SamplingWithFrameTrackComparisonReport(
+      Baseline<SamplingCounts> baseline_sampling_counts,
+      Baseline<orbit_client_data::ScopeStats> baseline_frame_track_stats,
+      Comparison<SamplingCounts> comparison_sampling_counts,
+      Comparison<orbit_client_data::ScopeStats> comparison_frame_track_stats,
+      absl::flat_hash_map<SFID, ComparisonResult> fid_to_comparison_results)
+      : baseline_sampling_counts_(std::move(baseline_sampling_counts)),
+        baseline_frame_track_stats_(std::move(baseline_frame_track_stats)),
+        comparison_sampling_counts_(std::move(comparison_sampling_counts)),
+        comparison_frame_track_stats_(std::move(comparison_frame_track_stats)),
+        fid_to_comparison_results_(std::move(fid_to_comparison_results)) {}
+
+  template <template <typename> typename Wrapper>
```

Comment:
This line confirms to me that you are way going too far with templates.

Reply:
Here -- I agree with you. The line is for an exercise here. Yet, the same line in `BaselineAndComparisonTmpl` serves an important [purpose](https://github.com/google/orbit/pull/3795#discussion_r900000659).